sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetPartitionDiscoverySuite.scala

```diff
@@ -969,4 +969,15 @@ class ParquetPartitionDiscoverySuite extends QueryTest with ParquetTest with Sha
       ))
     }
   }
+
+  test("SPARK-18108 Partition discovery fails with explicitly written long partitions") {
+    withTempPath { dir =>
```

Comment:
since this test is broken only when the vectorizedReader is on, I would also keep the conf here as enabled, just in case anyone changes the default implementation later.
so I would also wrap this as:
```scala
withSQLConf(SQLConf.PARQUET_VECTORIZED_READER_ENABLED.key -> "true") {
  withTempPath { dir =>
    ...
  }
}
```